### PR TITLE
Updating vector projection with helper function

### LINKED ISSUE #206: Update Vector plots with new GeoCAT-viz function
Update **NCL_overlay_6.py**, **NCL_vector_3.py**, **NCL_panel_1.py**, and **NCL_panel_13.py** with GeoCAT-viz utility function set_vector_density().

### PR DESCRIPTION
**Pull request by michaelavs**
closes #206

### CODE REVIEW

**Comment by michaelavs** [comment]
@anissa111 @hCraker I currently have this PR updating only one of the four plots Claire had outlined originally and  I wanted to get your opinions on only updating one of the four projections instead of all before converting this to a regular pull request. After looking through what the helper function was designed to replicate from NCL ([vcMinDistanceF](https://www.ncl.ucar.edu/Document/Graphics/Resources/vc.shtml)), I found only vector_3 used the NCL function and the other plots wouldn't necessarily benefit from adding the Python utility function. I did try to mess around and see if there would be any reasonable way to use the helper function in the other 3 projections, but it honestly would come down to adding extra lines of code that didn't necessarily change how the code ran or how the user defined the projection settings.

**Comment by anissa111** [comment]
> @anissa111 @hCraker I currently have this PR updating only one of the four plots Claire had outlined originally and I wanted to get your opinions on only updating one of the four projections instead of all before converting this to a regular pull request. After looking through what the helper function was designed to replicate from NCL ([vcMinDistanceF](https://www.ncl.ucar.edu/Document/Graphics/Resources/vc.shtml)), I found only vector_3 used the NCL function and the other plots wouldn't necessarily benefit from adding the Python utility function. I did try to mess around and see if there would be any reasonable way to use the helper function in the other 3 projections, but it honestly would come down to adding extra lines of code that didn't necessarily change how the code ran or how the user defined the projection settings.

Maybe I'm misunderstanding the purpose of the utility function, but is the utility function doing anything other than adding an extra line in `vector_3`? Additionally, when I run it locally, I'm getting a warning `UserWarning: Plot spacing is alrady greater or equal to 0.017'`, so I'm not certain it needs to be in this example, either.

It appears in `panel_1`, `panel_13`, and `overlay_6`, that the vector densities in the recreated plots are already consistent with the original NCL plots.

Was `set_vector_density` made to address a specific example other than recreating vcMinDistanceF?

**Comment by michaelavs** [comment]
> > @anissa111 @hCraker I currently have this PR updating only one of the four plots Claire had outlined originally and I wanted to get your opinions on only updating one of the four projections instead of all before converting this to a regular pull request. After looking through what the helper function was designed to replicate from NCL ([vcMinDistanceF](https://www.ncl.ucar.edu/Document/Graphics/Resources/vc.shtml)), I found only vector_3 used the NCL function and the other plots wouldn't necessarily benefit from adding the Python utility function. I did try to mess around and see if there would be any reasonable way to use the helper function in the other 3 projections, but it honestly would come down to adding extra lines of code that didn't necessarily change how the code ran or how the user defined the projection settings.
> 
> Maybe I'm misunderstanding the purpose of the utility function, but is the utility function doing anything other than adding an extra line in `vector_3`? Additionally, when I run it locally, I'm getting a warning `UserWarning: Plot spacing is alrady greater or equal to 0.017'`, so I'm not certain it needs to be in this example, either.
> 
> It appears in `panel_1`, `panel_13`, and `overlay_6`, that the vector densities in the recreated plots are already consistent with the original NCL plots.
> 
> Was `set_vector_density` made to address a specific example other than recreating vcMinDistanceF?

From my understanding, the function is to specify the space between the vector arrows on the projection. The issue for making this function was closed when Claire added this function, but I believe it was originally a community support issue and was asking if there was a function that would be similar to the vcMinDistanceF function from NCL. When I compared between the original python vector_3 and this version using the util function I also didn't see much of a change. But if I made the space larger, I did see a difference overall. So this is kind of where I'm not sure if it's really worth updating the code to use the function because it seems that matplotlib is able to handle the spacing itself and it comes out with distances consistent with the NCL version.

Also, I'm interested in the error you are getting because I am not getting it on my side when I run the code

**Comment by anissa111** [comment]
> From my understanding, the function is to specify the space between the vector arrows on the projection. The issue for making this function was closed when Claire added this function, but I believe it was originally a community support issue and was asking if there was a function that would be similar to the vcMinDistanceF function from NCL. When I compared between the original python vector_3 and this version using the util function I also didn't see much of a change. But if I made the space larger, I did see a difference overall. So this is kind of where I'm not sure if it's really worth updating the code to use the function because it seems that matplotlib is able to handle the spacing itself and it comes out with distances consistent with the NCL version.

Ahh, that makes sense, I understand.

> Also, I'm interested in the error you are getting because I am not getting it on my side when I run the code

This is interesting because also I do not see any difference in the plot when running the example with the util function and without.

**Comment by michaelavs** [comment]
> > From my understanding, the function is to specify the space between the vector arrows on the projection. The issue for making this function was closed when Claire added this function, but I believe it was originally a community support issue and was asking if there was a function that would be similar to the vcMinDistanceF function from NCL. When I compared between the original python vector_3 and this version using the util function I also didn't see much of a change. But if I made the space larger, I did see a difference overall. So this is kind of where I'm not sure if it's really worth updating the code to use the function because it seems that matplotlib is able to handle the spacing itself and it comes out with distances consistent with the NCL version.
> 
> Ahh, that makes sense, I understand.
> 
> > Also, I'm interested in the error you are getting because I am not getting it on my side when I run the code
> 
> This is interesting because also I do not see any difference in the plot when running the example with the util function and without.

I just realized that the spacing is going off degrees of latitude and not in something like inches (as I originally thought). That being said, I didn't see a real difference until I had the values of 'z' defined at `gvutil.set_vector_density(ds, 10)` because it is really such a small scale that the vector arrows are using for spacing. Let me know if that clears the error for you

**Comment by anissa111** [comment]
> I just realized that the spacing is going off degrees of latitude and not in something like inches (as I originally thought). That being said, I didn't see a real difference until I had the values of 'z' defined at `gvutil.set_vector_density(ds, 10)` because it is really such a small scale that the vector arrows are using for spacing. Let me know if that clears the error for you

Yeah, sure does!

File: Plots/Vectors/NCL_vector_3.py [hunk 1]
```diff
@@ -46,15 +44,17 @@
 
 # Generate axes using Cartopy projection
 ax = plt.axes(projection=ccrs.PlateCarree())
+z = gvutil.set_vector_density(ds, 0.017)
 
 # Draw vector plot
 # Notes
+
 # 1. We plot every third vector in each direction, which is not as nice as vcMinDistanceF in NCL
```
[the remark this answers is not in the document]
Comment (hCraker) [reply Plots/Vectors/NCL_vector_3.py:52]: When using `set_vector_density` we aren't explicitly stating that we want every third vector in both directions. Was there a reason for resolving this conversation without making any changes to this comment?

[the remark this answers is not in the document]
Comment (michaelavs) [reply Plots/Vectors/NCL_vector_3.py:52]: I had committed the changes that were suggested, but I think I didn't update the local version so when I pushed changes a couple days ago, this reverted back to original. I'll update locally now and push changes again.

**Comment by erogluorhan** [comment]
Please update this PR's "Draft" status if it is actually in a "Review in progress" status.

**Pull request merged**